radixdlt/src/main/java/com/radixdlt/consensus/PendingVotes.java

```diff
@@ -17,50 +17,100 @@
 
 package com.radixdlt.consensus;
 
+import com.google.common.collect.Maps;
 import com.google.inject.Inject;
 import com.radixdlt.consensus.validators.ValidationState;
 import com.radixdlt.consensus.validators.ValidatorSet;
 import com.radixdlt.crypto.ECDSASignature;
+import com.radixdlt.crypto.ECPublicKey;
 import com.radixdlt.crypto.Hash;
+import com.radixdlt.utils.Pair;
 
 import java.util.HashMap;
 import java.util.Objects;
 import java.util.Optional;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
 /**
  * Manages pending votes for various vertices
  */
 public final class PendingVotes {
-	private final HashMap<Hash, ValidationState> votes = new HashMap<>();
+	private static final Logger log = LogManager.getLogger();
+
+	private final HashMap<View, Pair<Hash, ValidationState>> state = Maps.newHashMap();

```

Comment:
Shouldn't confine ourselves to one view == one hash. As when we implement the better pacemaker, a leader collecting votes won't necessarily know what the correct hash for the view is.